mlflow/store/artifact/gcs_artifact_repo.py

```diff
@@ -18,17 +18,22 @@ class GCSArtifactRepository(ArtifactRepository):
     """
     Stores artifacts on Google Cloud Storage.
 
-    Assumes the google credentials are available in the environment,
-    see https://google-cloud.readthedocs.io/en/latest/core/auth.html.
+    :param artifact_uri: URI of GCS bucket
+    :param client: Optional, exposed for testing. Python module to use for GCS
+                   operations, defaults to google.cloud.storage.
+    :param gcs_client: Optional. The actual client to use for GCS operations; a default
+                       client object will be created if unspecified, using default
+                       credentials as described in https://google-cloud.readthedocs.io/en/latest/core/auth.html
     """
 
-    def __init__(self, artifact_uri, client=None):
+    def __init__(self, artifact_uri, client=None, gcs_client=None):
```

Comment:
Can we use the existing `client` argument?

Reply:
Good question - the existing `client` argument is exposed for tests to override/mock the GCS Python module used to instantiate a GCS client. I thought about repurposing it to allow specifying the actual storage client, but figured it wasn't worth making any breaking changes. That said, the GCS artifact repo isn't publicly documented, so I'm ok with updating to reuse `client` instead if you think it's preferable

Reply:
@smurching Thanks for the explanation!. Let's update it. Having two arguments for specifying a GCS client is really confusing.